mnelab/mainwindow.py

```diff
@@ -875,12 +875,14 @@ def auto_duplicate(self):
             self.model.duplicate_data()
             return True
         # otherwise ask the user
-        else:
-            msg = QMessageBox.question(self, "Overwrite existing data set",
-                                       "Overwrite existing data set?")
-            if msg == QMessageBox.No:  # create new data set
-                self.model.duplicate_data()
-                return True
+        msg = QMessageBox()
+        msg.setWindowTitle("Modify data set")
+        msg.setText("You are about to modify the current data set. How do you want to proceed?")  # noqa: E501
+        msg.addButton("Create new data set", QMessageBox.YesRole)
+        msg.addButton("Overwrite current data set", QMessageBox.NoRole)
+        if msg.exec() == 0:  # create new data set (button index 0)
```

Comment:
Instead of relying on indexes, would [this approach](https://doc.qt.io/qt-6/qmessagebox.html#advanced-usage) be better?

Also, what is the default now? I think it should be "Create new data set".

I think we should also disable the Esc key for now, because there is no Cancel option really. Cancelling should do nothing and basically not perform the requested action at all. However, this requires a little rework of the current workflow.

Reply:
> Instead of relying on indexes, would [this approach](https://doc.qt.io/qt-6/qmessagebox.html#advanced-usage) be better?

oh this is nice, thanks!

> Also, what is the default now? I think it should be "Create new data set".

It is "Create new data set."

> I think we should also disable the Esc key for now, because there is no Cancel option really. 

I couldn't find a way to disable <kbd>Esc</kbd>, do you know how this can be done?

> Cancelling should do nothing and basically not perform the requested action at all. However, this requires a little rework of the current workflow.

Maybe all the dialogs for actions which can create or overwrite a dataset could have a checkbox "Overwrite current data set" instead of the extra dialog?

Reply:
> I couldn't find a way to disable <kbd>Esc</kbd>, do you know how this can be done?

Hm, not sure - can you pass `None` to [`setEscapeButton`](https://doc.qt.io/qt-6/qmessagebox.html#setEscapeButton)?

Reply:
If this doesn't work I think we have to role our own `QDialog` object and [overwrite the `keyPressEvent`](https://stackoverflow.com/a/3127786/1112283). It probably also works by overwriting the method for our existing `msg`. But we don't need to do it now if it's too much work.

Reply:
> Maybe all the dialogs for actions which can create or overwrite a dataset could have a checkbox "Overwrite current data set" instead of the extra dialog?

Yes, I don't know, this would mean that we need to add that checkbox to almost all dialogs. I would prefer a separate dialog. But we could delay the action until the second dialog has been accepted (either with creating a new data set or overwriting the current one); if it gets rejected (cancelled) then we don't perform the action.

Reply:
> Hm, not sure - can you pass `None` to [`setEscapeButton`](https://doc.qt.io/qt-6/qmessagebox.html#setEscapeButton)?

doesn't help :/


> Yes, I don't know, this would mean that we need to add that checkbox to almost all dialogs. 

I guess this could be simplified by creating a mixin which some dialogs derive from 🤔 

> But we could delay the action until the second dialog has been accepted (either with creating a new data set or overwriting the current one); if it gets rejected (cancelled) then we don't perform the action.

That would require to check for a return from (or exception raised by) `auto_duplicate` in those functions, right?